src/App.js

```diff
@@ -1,19 +1,59 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog'
+
 
 const App = () => {
+  
+  const senderSet = new Set(chatMessages.map((entry) => entry.sender));
+  const senderNames = (Array.from(senderSet));
+  
+  const [chatMessageData, setChatMessagData] = useState(chatMessages);
+  const[likeCount, setLikeCount] = useState(0);
+
+  const updateEntryData = updatedEntry => {
+    const entries = chatMessageData.map(entry => {
+      if (entry.id === updatedEntry.id){
+        return updatedEntry;
+      } 
+      else{
+        return entry;
+      }
+    });
+    const likedCount = entries.filter(entry => entry.liked).length;
+    setChatMessagData(entries);
```

Comment:
👀 Since the next version of the chat data depends on the current version, prefer using the callback style of setChatMessagData:
```javascript
setChatMessagData(prevEntries => ( 
    // logic to use current chats to make a new copy with the desired liked status flipped (the map logic above) 
    // return the new chats to update the state 
))